p2p/starknet/client.go

```diff
@@ -0,0 +1,129 @@
+package p2p
+
+import (
+	"context"
+
+	p2p_spec "github.com/NethermindEth/juno/p2p/starknet/p2p-spec"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/protocol"
+	"google.golang.org/protobuf/proto"
+)
+
+type NewStreamFunc func(ctx context.Context, pids ...protocol.ID) (network.Stream, error)
+
+type StarknetClient struct {
+	newStream  NewStreamFunc
+	protocolID protocol.ID
+	log        utils.Logger
+}
+
+func NewStarknetClient(newStream NewStreamFunc, protocolID protocol.ID, log utils.Logger) (*StarknetClient, error) {
+	return &StarknetClient{
+		newStream:  newStream,
+		protocolID: protocolID,
+		log:        log,
+	}, nil
+}
+
+func (c *StarknetClient) sendAndReceiveInto(ctx context.Context, req, res proto.Message) error {
+	stream, err := c.newStream(ctx, c.protocolID)
+	if err != nil {
+		return err
+	}
+	defer stream.Close() // todo: dont ignore close errors
+
+	reqBytes, err := proto.Marshal(req)
+	if err != nil {
+		return err
+	}
+
+	_, err = stream.Write(reqBytes)
```

Comment:
Keep in mind, pathfinder's implementation zstd compress the request or response. No idea what starknet wants. For reference ethereum snappy compress the stream.

Reply:
noted, thanks!